akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -80,8 +81,60 @@ class GraphStageLogicSpec extends AkkaSpec with GraphInterpreterSpecKit with Con
     }
   }
 
+  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```

Comment:
I always use `final` as a default.
And since this is a test, adding `case ` in front of the class to save typing `new ` all over the place below saves me a bit of typing.